tests/ui/result_map_or_into_option.rs

```diff
@@ -0,0 +1,14 @@
+// run-rustfix
+
+#![warn(clippy::result_map_or_into_option)]
+
+fn main() {
+    let opt: Result<u32, &str> = Ok(1);
+    let _ = opt.map_or(None, Some);
```

Comment:
Please also add a test for `opt.map_or(None, |x| Some(x))`, which also should get linted and one for `opt.map_or(None, |_x| Some(y))`, which should not get linted.

Reply:
Partially fixed in fb276dc3fa1d644ce6de1849ff3b78d3e54f5d9e

It looks like `opt.map_or(None, |x| Some(x))` is caught by `clippy::redundant-closure`
```
error: redundant closure found
  --> $DIR/result_map_or_into_option.rs:17:22
   |
LL |     opt.map_or(None, |x| Some(x));
   |                      ^^^^^^^^^^^ help: remove closure as shown: `Some`
   |
   = note: `-D clippy::redundant-closure` implied by `-D warnings`
```

Should I still add that test?

Reply:
No then you don't have to lint this case. Good catch!